typescript/core/src/ChunkCursor.ts

```diff
@@ -211,80 +160,75 @@ export class ChunkCursor {
         continue;
       }
 
-      result.record.records.sort(([logTimeA], [logTimeB]) => Number(logTimeA - logTimeB));
-      if (reverse) {
-        // If we used `logTimeB - logTimeA` as the comparator for reverse iteration, messages with
-        // the same timestamp would not be in reverse order. To avoid this problem we use reverse()
-        // instead.
-        result.record.records.reverse();
-      }
-
-      for (const [logTime] of result.record.records) {
-        if (logTime < this.chunkIndex.messageStartTime) {
-          throw new Error(
-            `Encountered message index entry in channel ${result.record.channelId} with logTime (${logTime}) earlier than chunk messageStartTime (${this.chunkIndex.messageStartTime}) in chunk at offset ${this.chunkIndex.chunkStartOffset}`,
-          );
-        }
-        if (logTime > this.chunkIndex.messageEndTime) {
-          throw new Error(
-            `Encountered message index entry in channel ${result.record.channelId} with logTime (${logTime}) later than chunk messageEndTime (${this.chunkIndex.messageEndTime}) in chunk at offset ${this.chunkIndex.chunkStartOffset}`,
-          );
-        }
-      }
-
-      let startIndex = 0;
-      if (reverse) {
-        if (this.#endTime != undefined) {
-          startIndex = sortedIndexBy(result.record.records, this.#endTime, (logTime) => -logTime);
-        }
-      } else {
-        if (this.#startTime != undefined) {
-          startIndex = sortedIndexBy(result.record.records, this.#startTime, (logTime) => logTime);
-        }
-      }
-
-      if (startIndex >= result.record.records.length) {
-        continue;
-      }
-      if (reverse) {
-        if (
-          this.#startTime != undefined &&
-          result.record.records[startIndex]![0] < this.#startTime
-        ) {
-          continue;
-        }
-      } else {
-        if (this.#endTime != undefined && result.record.records[startIndex]![0] > this.#endTime) {
-          continue;
-        }
-      }
-
-      this.#messageIndexCursors.push({
-        index: startIndex,
-        channelId: result.record.channelId,
-        records: result.record.records,
-      });
+      this.#orderedMessageOffset = this.#orderedMessageOffset.concat(result.record.records);
```

Comment:
Consider using `push` rather than `concat` for better performance.

It's possible that using `.flat()` on an array of arrays would be even faster depending on how it is implemented.